src/project_model.rs

```diff
@@ -133,6 +138,31 @@ impl Package {
 pub struct RacerProjectModel(pub Arc<ProjectModel>);
 
 impl racer::ProjectModelProvider for RacerProjectModel {
+    fn edition(&self, manifest: &Path) -> Option<racer::Edition> {
+        self.0
+            .package_for_manifest(manifest)
+            .map(|pkg| self.0.get(pkg).edition)
+    }
+
+    fn search_dependencies(
+        &self,
+        manifest: &Path,
+        search_fn: Box<dyn Fn(&str) -> bool>,
+    ) -> Vec<(String, PathBuf)> {
+        let pkg = match self.0.package_for_manifest(manifest) {
+            Some(pkg) => pkg,
+            None => return vec![],
+        };
+
+        pkg.deps(&self.0)
+        .iter()
+        .filter(|d| search_fn(&d.crate_name))
+        .filter_map(|d| self.0.get(d.pkg).lib.as_ref())
+        .cloned()
+        .map(|(src_path, crate_name)| (crate_name, src_path))
+        .collect()
```

Comment:
@kngwyu I tried to match Racer's own implementation; does this look correct?

Reply:
Yeah looks good :+1:

Reply:
Thanks for taking a look!